packages/edit-site/src/components/editor/global-styles-provider.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useEffect } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { getGlobalStyles } from './global-styles-renderer';
+
+const GlobalStylesContext = createContext( {
+	/* eslint-disable no-unused-vars */
+	getProperty: ( context, family, name, units ) => {},
```

Comment:
Jorge and I talked about this today: the idea would be to make the global styles provider unit-agnostic. At the moment, it requires to move the unit processing to the sidebar component because UI controls can't absorb that. A potential next step would be whether the controls can also absorb that processing, but that's a separate discussion from this PR.

Reply:
Note that, at the moment, the font size values are serialized as px values (see [serializer](https://github.com/WordPress/gutenberg/blob/master/packages/element/src/serialize.js#L336)) in the block level (how post editor works). We follow the same at the global level, so I went for the simple approach (just toPx/fromPx functions) instead of trying to cover any other potential unit. When done, this should be addressed at both the block and global level.

https://github.com/WordPress/gutenberg/pull/24250/commits/cbcefd2d17f846825d8e86517575efabb487d160

Reply:
Created https://github.com/WordPress/gutenberg/issues/25137 to follow-up on this.